workflow-to-js-transformer/src/main/java/com/github/cafdataprocessing/workflow/transform/TransformerFunctions.java

```diff
@@ -42,9 +52,31 @@ public static String getEnvironmentValue(final String name)
      * @return Queue value associated with specified worker name or null if no match found.
      */
     public static String getWorkerQueueFromEnvironment(final String workerName){
-        if(workerName==null || workerName.isEmpty()){
+        if (workerName == null || workerName.isEmpty()) {
             return null;
         }
         return getEnvironmentValue(workerName.toLowerCase(Locale.ENGLISH) + ".taskqueue");
     }
+
+    public static String getTenantSpecificConfigValue(final String apiClientBaseUrl, final String tenantId, final String key)
+        throws DocumentWorkerTransientException
+    {
+        final ApiClient apiClient = new ApiClient();
+        apiClient.setBasePath(apiClientBaseUrl);
+        final TenantsConfigurationApi tenantsApi = new TenantsConfigurationApi(apiClient);
+        try {
+            final EffectiveTenantConfigValue effectiveTenantConfigValue = tenantsApi.getEffectiveTenantConfig(tenantId, key);
+            LOG.debug("Retrieved value for tenant configuration using key: {}", key);
+            LOG.debug("Retrieved value for tenant configuration is of type: {}", effectiveTenantConfigValue.getValueType());
+            return effectiveTenantConfigValue.getValue();
+        } catch (final ApiException ex) {
+            if (ex.getCode() == 500) {
+                LOG.error("Unable to obtain tenant configuration from processing service for tenant: {} using key: {} as the service could"
```

Comment:
I'd change this to check for the 404 first since we know that specific indicates failure to find the configuration. A 500 could be more than just not being able to contact the service, it could be due to the database being down, an internal JavaScript error etc, so we shouldn't be so specific in the log message.